test/xpu_api/numerics/c.math/cmath.pass.cpp

```diff
@@ -240,20 +240,23 @@ void test_isinf()
     static_assert((std::is_same_v<decltype(dpl::isinf((float)0)), bool>));
     static_assert((std::is_same_v<decltype(dpl::isinf(0)), bool>));
 
-    auto fnc = []()
+    if constexpr (HasDoubleSupportInRuntime{})
     {
-        typedef decltype(dpl::isinf((double)0)) DoubleRetType;
+        auto fnc = []()
+        {
+            typedef decltype(dpl::isinf((double)0)) DoubleRetType;
 #if !defined(__linux__) || defined(__clang__)
-        static_assert((std::is_same_v<DoubleRetType, bool>));
+            static_assert((std::is_same_v<DoubleRetType, bool>));
 #else
-        // GLIBC < 2.23 defines 'isinf(double)' with a return type of 'int' in
-        // all C++ dialects. The test should tolerate this when libc++ can't work
-        // around it.
-        // See: https://sourceware.org/bugzilla/show_bug.cgi?id=19439
-        static_assert((std::is_same_v<DoubleRetType, bool> || std::is_same_v<DoubleRetType, int>));
+            // GLIBC < 2.23 defines 'isinf(double)' with a return type of 'int' in
+            // all C++ dialects. The test should tolerate this when libc++ can't work
+            // around it.
+            // See: https://sourceware.org/bugzilla/show_bug.cgi?id=19439
+            static_assert((std::is_same_v<DoubleRetType, bool> || std::is_same_v<DoubleRetType, int>));
 #endif
-    };
-    IF_DOUBLE_SUPPORT_L(fnc)
+        };
+        fnc();
```

Comment:
Do we really need to define the lambda and immediately execute it? It seems like we can perform all checks inside the `if constexpr` body.

Similar simplification can be applied in other places.

Reply:
it's required, because otherwise we have compile error (in case when we place old macro argument simple under `constexpr if`

Reply:
I don't understand why there should be any compilation issues. What makes the lambda necessary?

I have tried manually with abs1.pass and it works.

Reply:
Fixed

Reply:
Fixed